web/src/javascript/app.js

```diff
@@ -3,21 +3,37 @@ import {drawGame, createGameCanvas} from "./game";
 import {gameRefresh, leaderboardRefresh} from "./constants";
 
 const gameCanvas = createGameCanvas();
-let currentState;
 
-(new Worker("javascript/gameWebWorker.bundle.js")).onmessage = message => currentState = message.data;
+let currentState;
+const networkWorker = new Worker("javascript/gameWebWorker.bundle.js");
+networkWorker.onmessage = message => {
+  // This is to prevent Chrome's GC from deleting the worker.
+  // It's happening on Chrome but not on FF.
+  if(!networkWorker) {
+    console.error("Got GCed");
+  }
+  currentState = message.data;
+};
 
-async function updateLoop() {
+async function updateGame() {
+  const startTime = (new Date()).getTime();
   if(currentState) {
     drawGame(currentState, gameCanvas);
   }
+
+  const elapsed = (new Date()).getTime() - startTime;
+  setTimeout(updateGame, 1000/gameRefresh - elapsed);
```

Comment:
Close, but not exactly. It's to make sure that the `updateGame` is calles regularly (always the same frequence).